servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DefaultDnsClient.java

```diff
@@ -846,6 +847,10 @@ private void handleResolveDone0(final Future<DnsAnswer<T>> addressFuture,
                             cancellableForQuery = null;
                         }
                         try {
+                            // Shuffle events to avoid multiple clients connecting to the same upstream IP address if
+                            // they receive identical result set from DNS.
+                            Collections.shuffle(events);
```

Comment:
I was considering adding a builder option to let users opt-out of shuffling, but then I couldn't find a good reason why they may need a sorted list.
Worst case, they can wrap SD interface and apply `map` on the publisher that will sort again.